pkg/workceptor/kubernetes.go

```diff
@@ -256,17 +255,15 @@ func (kw *kubeUnit) kubeLoggingWithReconnect(streamWait *sync.WaitGroup, stdout
 				}
 
 				return
-			} else if _, ok := err.(http2.GoAwayError); ok {
-				// GOAWAY is sent by the server to indicate that the server is gracefully shutting down
-				// this happens if the kube API server we are connected to is being restarted or is shutting down
-				// for example during a cluster upgrade and rolling restart of the master node
+			} else if err != nil {
```

Comment:
now that we expend the err handling i think the `if err == io.EOF` can also be folded in to `err != nil` and this logic can be restructured a bit

Reply:
Thanks for this! All folded into one `err != nil`